### ISSUE TITLE
double clicking on a word selects whole line instead

### ISSUE THREAD

**Issue by Makombo** [notepadqq/notepadqq#222]
is it just me, or this is not quite working?

If I have a sentence on line 1 say:
foo bar

and I double click foo.. I expect only foo to be highlighted not both foo an bar

is this perhaps a setting I need to update?
Thanks

**Comment by guekho64** [comment]
Maybe you're running an outdated Qt version.

That happens to me whenever I try to install Notepadqq without installing Qt 5.5.1 first (That's my favourite version)

I recommend you to delete your local Notepadqq config to avoid strange behaviour once you install an updated release of Qt.

Just do:

rm -r "${HOME}/.config/Notepadqq/"

Or delete it using your favourite graphical file manager, like Dolphin or Nemo.

**Comment by darealshinji** [comment]
Same behavior on Qt 5.2.1. But to be honest this is what I expect and got used to. Maybe this should be a setting so the user can decide whether it highlights a word, a line or nothing at all.

**Comment by Makombo** [comment]
No, this is not standard behavior. in every popular application out there
that can edit text line by line e.g. msword, notepad++, email clients... if
you double click, they select the word *but* if you triple click then the
whole line is selected ... no point in breaking conventions that are
standard across platforms.

On Tue, Jun 7, 2016, 5:26 AM darealshinji <notifications@github.com> wrote:

> Same behavior on Qt 5.2.1. But to be honest this is what I expect and got
> used to. Maybe this should be a setting so the user can decide whether it
> highlights a word, a line or nothing at all.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/notepadqq/notepadqq/issues/222#issuecomment-224240834>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFQFYOFo_rdtrgG9fv8x8fa_kBo2AHbBks5qJUdkgaJpZM4Iu25G>
> .
>
-- 

Somo

**Comment by guekho64** [comment]
Maybe a specifical config in your OS?

Here, it's working as you expect. 

Using Notepadqq 0.51 with Qt 5.5.1 on Ubuntu OS with LXDE Desktop.

**Comment by darealshinji** [comment]
Double clicking to select a word works fine after I installed Qt 5.6 into my home directory.

**Comment by danieleds** [comment]
Working as expected here (double-clicking selects just the word).
It might be indeed related to the old Qt libraries.

**Comment by Makombo** [comment]
I am using linux mint.. 
for some reason it will not detect that I have Qt 5.6 already instally in ~/Qt

I tried to uninstall notepadqq and notepadqq-common throught the software manager  and it's not working

is there a way to forcefully remove the underlying Qt 5.2.1 and make note pad point to the ./bin and ./lib files of Qt 5.6??

**Comment by darealshinji** [comment]
@Makombo you must edit the startup script `/usr/lib/notepadqq/notepadqq.sh` to find Qt 5.6.

**Comment by Makombo** [comment]
@darealshinji  Thanks this worked. The issue was that I was using the PPA version of notepadqq which apparently doesn't yet support the latest Qt5.6  (only upto 5.5 was supported in that script). In future a generic script that just looks for the latest available Qt version that is atleast 5.3 needs to written.

**Comment by darealshinji** [comment]
@Makombo That's a good idea. I guess something like this would do it:
``` sh
for n in `seq 3 9`
do
  PATH="~/Qt/5.$n;$PATH"
done
export $PATH
```

**Comment by Makombo** [comment]
Yeah well. What you have here is essentially the same you have added three
more future versions.. what happens when qt moves to 6.0??

What I was thinking instead was along the lines of: rather than guessing
what qt version you *might*,  the script should figure out what you qt
version you *actually* have and then deciding whether that version meets
the minimum requirements e.g. it might use qmake -v command and extract
relevant info from that .

But I really don't know much about qt or Linux scripting.. so just throwing
this out there .
Thanks again... job well done to those that have contributed to coding
notepadqq up!

On Thu, Jun 9, 2016, 6:02 AM darealshinji <notifications@github.com> wrote:

> @Makombo <https://github.com/Makombo> That's a good idea. I guess
> something like this would do it:
>
> for n in `seq 3 9`do
>   PATH="~/Qt/5.$n;$PATH"doneexport $PATH
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/notepadqq/notepadqq/issues/222#issuecomment-224863549>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFQFYDXL5LNL0jvMrrJlOlsKNvTk6eP9ks5qJ_KvgaJpZM4Iu25G>
> .
>
-- 

Somo

**Comment by darealshinji** [comment]
> what happens when qt moves to 6.0??

Qt4 programs can't make use of Qt5 libraries, I'm sure the same would apply whenever Qt6 came out.

**Comment by guekho64** [comment]
Are you saying that, due to the fact that Notepadqq uses Qt 5.x, when Qt Devs release v6.x, you'll need to rewrite entire Notepadqq ( or Partially)?

**Comment by darealshinji** [comment]
@guekho64 You will very likely have to change some code to make it possible to build the program with v6 (might not be much at all, depending on the program). This is at least the case if you want to build a program with v5 that was written for v4 and vice versa.
And a compiled program can only use the Qt libraries it was linked against. Let's say a program was linked against `libQt5Core.so.5`, it won't even attempt to load its Qt4 conterpart `libQtCore.so.4`. And even if you rename or manipulate the files to force it using the Qt4 variant the program will still not work.

**Comment by Makombo** [comment]
Well you will definitely need to atleast add the few lines of code to the
same startup script that I did, in order to get 5.6 working... I am not
sure whether there would also some other files affected

On Thu, Jun 9, 2016, 5:57 PM guekho64 <notifications@github.com> wrote:

> Are you saying that, due to the fact that Notepadqq uses Qt 5.x, when Qt
> Devs release v6.x, you'll need to rewrite entire Notepadqq ( or Partially)?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/notepadqq/notepadqq/issues/222#issuecomment-225051268>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFQFYHPYhKh6nsH0Widnpf5cXzrE6GpVks5qKJptgaJpZM4Iu25G>
> .
>
-- 

Somo

**Comment by guekho64** [comment]
If I'm right, recently I suggested Notepadqq devs to add some lines of code to the Notepadqq startup script.

I dont't know if they've alredy added those lines to the "Official" release from the PPA of Trusty or Xenial, because of two reasons:

*I modified my local script since I discovered that "Bug" with Qt 5.6
*Since Qt 5.6, Notepadqq started to behave incorrectly with some usual Keyboard Shortcuts, like "Find" (Ctrl + H) or "Replace" (Ctrl + R), so I decided to stay with Qt 5.5.1

That's why I don't know if the "Qt 5.6 compatible startup script" is now in the PPA release

PD: Theoricaly, all Qt 5.x releases SHOULD work flawlessly with Notepadqq, but here we have Qt 5.6 with those buggy keyboard shortcuts. People keep saying that Qt is the one causing so much trouble with shortcuts in other apps.

But ahead from that, it shouldn't be necessay to modify the core structure of Notepadqq as long as you use Qt 5.x
